### ISSUE TITLE
Previously created ViewModels don't go through the lifecycle in cached fragments

### ISSUE THREAD

**Issue by vvolkgang** [MvvmCross/MvvmCross-AndroidSupport#108]
_TL;DR;_ VM Lifecycle is only ran when the cached fragment doesn't already have an instance of a VM. There seems to be no functionality to run it when the Vm is already created.

First, in `MvxCachingFragmentCompatActivity.ShowFragment`, when you have a cached fragment, you need to set the received bundle as the `Arguments` prop. I wasn't doing that.

Second, VM Init is only called when the fragment doesn't have one already created, in [MvxFragmentExtensions.LoadViewModel(...)](https://github.com/MvvmCross/MvvmCross-AndroidSupport/blob/master/Cirrious.MvvmCross.Droid.Support.Fragging/Fragments/MvxFragmentExtensions.cs#L77).

If theres an instance of a VM attached to the fragment, it will be null checked to stop the execution in two places:
1. [MvxFragmentExtensions.OnCreate(...)](https://github.com/MvvmCross/MvvmCross-AndroidSupport/blob/master/Cirrious.MvvmCross.Droid.Support.Fragging/Fragments/MvxFragmentExtensions.cs#L35)
2. [MvxViewExtensionMethods.OnViewCreate](https://github.com/MvvmCross/MvvmCross/blob/4.0/Cirrious/Cirrious.MvvmCross/Views/MvxViewExtensionMethods.cs#L25)

The current `ViewModel` lifecycle seems to be handled by `MvxDefaultViewModelLocator`, thus, a simple solutions seems to be to improve `MvxDefaultViewModelLocator` with a `Load(IMvxViewModel vm, IMvxBundle parameterValues, IMvxBundle savedState)`, that skips the VM instantiation and only calls the lifecycle with the new params.

Do you have any other idea @slodge @Cheesebaron @martijn00 ?

**Comment by vvolkgang** [comment]
Forgot to mention the last step to the solution. After adding that method to `IMvxDefaultViewModelLocator`, change this [null check](https://github.com/MvvmCross/MvvmCross-AndroidSupport/blob/master/Cirrious.MvvmCross.Droid.Support.Fragging/Fragments/MvxFragmentExtensions.cs#L37) to run the VM lifecycle with the existing instance instead.

**Comment by vvolkgang** [comment]
Created a PR in MvvmCross for ViewModel reloading.
